src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -767,6 +754,19 @@ def __call__(
                 # compute the previous noisy sample x_t -> x_t-1
                 latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs, return_dict=False)[0]
 
+                if callback_on_step_end is not None:
+                    callback_kwargs = {}
+                    for k in callback_on_step_end_inputs:
+                        callback_kwargs[k] = locals()[k]
+                    callback_kwargs = callback_on_step_end(i, t, callback_kwargs)
+
+                    latents = callback_kwargs.pop("latents", latents)
+                    guidance_scale = callback_kwargs.pop("guidance_scale", guidance_scale)
+                    prompt_embeds = callback_kwargs.pop("prompt_embeds", prompt_embeds)
+                    cross_attention_kwargs = callback_kwargs.pop("cross_attention_kwargs", cross_attention_kwargs)
+                    guidance_rescale = callback_kwargs.pop("guidance_rescale", guidance_rescale)
+                    do_classifier_free_guidance = guidance_scale > 1.0
+
                 # call the callback, if provided
```

Comment:
This comment should be:
`# call the callback_on_step_end, if provided`

and it should be moved to before the `if callback_on_step_end is not None:` on line 757